Sources/Target/POSIX/ELFProcess.cpp

```diff
@@ -203,10 +201,11 @@ EnumerateLinkMap(ELFProcess *process, Address addressToDPtr,
     if (error != ds2::kSuccess)
       return error;
 
+    shlib.main = (process->loadBase() == linkMap.baseAddress);
```

Comment:
This is not what I had in mind when we talked about it. linkMap.baseAddress is the relocation applied to the executable, and it will be zero for non-PIE executables. loadBase seems to be the lowest virtual address that is mapped in the process. Are you sure this is working for non-PIE executables?